brainiak/reprsimil/brsa.py

```diff
@@ -891,6 +917,29 @@ def _calc_dist2_GP(self, coords=None, inten=None,
         return dist2, inten_diff2, space_smooth_range, inten_smooth_range,\
             n_smooth
 
+    def _calc_tau2(self, log_SNR_invK_tilde_log_SNR, n_V):
+        """ Calculate the MAP estimation of the standar deviation of GP
+            on log(SNR)
+            Note that the form of the MAP estimate
+            of tau2 depends on the form of prior imposed.
+        """
+        if self.tau2_prior == 'halfCauchy':
+            tau2 = (log_SNR_invK_tilde_log_SNR - n_V * self.tau_range**2
+                    + np.sqrt(n_V**2 * self.tau_range**4 + (2 * n_V + 8)
+                              * self.tau_range**2
+                              * log_SNR_invK_tilde_log_SNR
+                              + log_SNR_invK_tilde_log_SNR**2))\
+                / 2 / (n_V + 2)
+            log_ptau = scipy.stats.halfcauchy.logpdf(
+                tau2**0.5, scale=self.tau_range)
+        else:
+            tau2 = (log_SNR_invK_tilde_log_SNR + 2 * self.tau_range**2) /\
+                (2 * 2 + 2 + n_V)
+            log_ptau = scipy.stats.invgamma.logpdf(
+                tau2, scale=self.tau_range**2, a=2)
+        # LL_inc is  p(tau2)
+        return tau2, log_ptau
+
```

Comment:
Exactly. You document their usage just like you document the usage of the current strings, except that the code now is easy to inspect and understand. Furthermore, it is easy to add new functionality by adding a new function, instead of adding code to a single big function that behaves according to a string parameter.